### DISCUSSION TITLE
Customize the dashboard based on the physical device running HA

### DISCUSSION THREAD

**Discussion by fredck** [home-assistant/frontend#16375] [Dashboard]
I need to be able to identify the physical device running HA (e.g. a specific phone running the companion app) so the dashboard can be customized to that device.

## Use case

Display a room specific card based on the room the device is located at (ESPresence), so the user has the room controls straight when opening the app. How cool is that?!

## Research so far

I've been running and inspecting the iPhone companion app using the Xcode Simulator but I couldn't find anything that's useful to identify the device.

The following would be enough for that purpose since it needs to be unique in a HA installation anyway:

![image](https://user-images.githubusercontent.com/630060/235688050-064cb553-044f-4a54-9308-bafaf23babfa.png)

Since the information is already available to the companion app, it should be theoretically cheap to expose it to the frontend. If only one would know how to do that! 😅

**Comment by ildar170975** [comment]
Meanwhile you may try using custom cards like state-switch, layout-card.

**Reply by fredck** [reply]
> Meanwhile you may try using custom cards like state-switch, layout-card.

Thanks for the tip. Once I figure out a way to identify the device (browser_mod maybe, although it does way too much more), that's definitely the way to go ;)

**Reply by ildar170975** [reply]
You may try using mediaquery instead of device ID in some cases - no need to install browser_mod then.
